Fix gen_call_with_extracted_types for literals

tkf opened the pull request:
Prior to this patch, executing (say) `@less 1` in REPL prints a cryptic message `ERROR: LoadError: type Int64 has no field head` although there is a branch in `gen_call_with_extracted_types` which is supposed to handle this case.

Before:

```julia
julia> @less 1
ERROR: LoadError: type Int64 has no field head
Stacktrace:
 [1] getproperty(::Any, ::Symbol) at ./sysimg.jl:18
 [2] gen_call_with_extracted_types(::Module, ::Symbol, ::Int64) at /.../julia/usr/share/julia/stdlib/v0.7/InteractiveUtils/src/macros.jl:58
 [3] @less(::LineNumberNode, ::Module, ::Any) at /.../julia/usr/share/julia/stdlib/v0.7/InteractiveUtils/src/macros.jl:70
in expression starting at REPL[1]:1
```

After:
```julia
julia> @less 1
ERROR: expression is not a function call or symbol
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at none:0
```

tkf committed:
Fix gen_call_with_extracted_types for literals

Prior to this patch, executing (say) `@less 1` in REPL prints a
cryptic message `ERROR: LoadError: type Int64 has no field head`
although there is a branch in `gen_call_with_extracted_types` which is
supposed to handle this case.

staticfloat commented on stdlib/InteractiveUtils/src/macros.jl:
```diff
@@ -32,7 +32,7 @@ function gen_call_with_extracted_types(__module__, fcn, ex0)
     exret = Expr(:none)
     if !isa(ex, Expr)
-        exret = Expr(:call, :error, "expression is not a function call or symbol")
+        return Expr(:call, :error, "expression is not a function call or symbol")
```
Let's just separate this `if` from the rest of the `elseif` chain.  E.g.:

```julia
if !isa(ex, Expr)
    return Expr(:call, :error, "expression is not a function call or symbol")
end

if ex.head == :call
....
```

That makes it a little clearer that this `if` statement is conceptually at a slightly different level than the next couple of `if` statements.

tkf committed:
Reorder if blocks for readability

tkf commented:
@staticfloat Thank you! That's a good point. Fixed, added a bit more readability tweaks, and pushed.

staticfloat reviewed (approved):
Looks good to me!  Thanks!

fredrikekre commented on stdlib/InteractiveUtils/test/runtests.jl:
```diff
@@ -331,6 +331,16 @@ if Sys.ARCH === :x86_64 || occursin(ix86, string(Sys.ARCH))
+        @test exret.head == :call
+        @test exret.args[1] == :error
+    end
+end
```
Perhaps better to test what caused the failure in the first place instead of this internal function? I think a test like
```julia
@test_throws ErrorException("expression is not a function call or symbol") @less 1
```
might be better?

tkf committed:
Test user-facing API directly

tkf commented:
@fredrikekre Yes, it makes sense. Thanks. Fixed it.

(I switched to `@code_lowered` instead of `@less` to avoid accidentally launching `less` when someone changes how InteractiveUtils macro work in the future.)
